lib/maze/servlets/command_servlet.rb

```diff
@@ -8,25 +8,69 @@ module Servlets
     # Allows clients to queue up "commands", in the form of Ruby hashes, using Maze::Server.commands.add.  GET
     # requests made to the /command endpoint will then respond with each queued command in turn.
     class CommandServlet < BaseServlet
+
+      NOOP_COMMAND = '{"action": "noop", "message": "No commands queued"}'
+
       # Serves the next command, if these is one.
       #
-      # @param _request [HTTPRequest] The incoming GET request
+      # @param request [HTTPRequest] The incoming GET request
       # @param response [HTTPResponse] The response to return
-      def do_GET(_request, response)
+      def do_GET(request, response)
+
+        if request.query.empty?
+          # Non-idempotent mode - return the "current" command
+          commands = Maze::Server.commands
+
+          if commands.size_remaining == 0
+            response.body = NOOP_COMMAND
+            response.status = 200
+          else
+            command = commands.current
+            response.body = JSON.pretty_generate(command)
+            response.status = 200
+            commands.next
+          end
+        else
+          $logger.info "request.query = #{request.query}"
+
+
+          # Idempotent mode
+          after_uuid = request.query['after']
+          if after_uuid.nil?
```

Comment:
Is `after` intended to be the _only_ recognised query parameter?  If not, can we rename the `return_command` method to be indicative that it returns the command after the uuid?

Reply:
Renamed to `command_after(uuid,...`.